src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs

```diff
@@ -118,33 +118,42 @@ public void TFS_470021()
     </xs:complexType>
   </xs:element>
 </xs:schema>";
-            using (XmlWriter w = XmlWriter.Create("cham.xsd"))
+
+            using (var tempDirectory = new TempDirectory())
             {
-                using (XmlReader r = XmlReader.Create(new StringReader(cham)))
-                    w.WriteNode(r, true);
-            }
-            XmlSchemaSet ss = new XmlSchemaSet();
-            ss.ValidationEventHandler += new ValidationEventHandler(ValidationCallback);
+                string chamPath = Path.Combine(tempDirectory.Path, "cham.xsd");
+                string tempDirectoryPath = tempDirectory.Path[tempDirectory.Path.Length - 1] == Path.DirectorySeparatorChar ?
+                    tempDirectory.Path :
+                    tempDirectory.Path + Path.DirectorySeparatorChar;
 
-            ss.Add(null, XmlReader.Create(new StringReader(cham)));
-            ss.Add(null, XmlReader.Create(new StringReader(main)));
-            ss.Compile();
+                using (XmlWriter w = XmlWriter.Create(chamPath))
+                {
+                    using (XmlReader r = XmlReader.Create(new StringReader(cham)))
+                        w.WriteNode(r, true);
+                }
+                XmlSchemaSet ss = new XmlSchemaSet();
+                ss.ValidationEventHandler += new ValidationEventHandler(ValidationCallback);
 
-            Assert.Equal(ss.Count, 2);
-            foreach (XmlSchemaElement e in ss.GlobalElements.Values)
-            {
-                _output.WriteLine(e.QualifiedName.ToString());
-                XmlSchemaComplexType type = e.ElementSchemaType as XmlSchemaComplexType;
-                XmlSchemaSequence seq = type.ContentTypeParticle as XmlSchemaSequence;
-                foreach (XmlSchemaObject child in seq.Items)
+                ss.Add(null, XmlReader.Create(new StringReader(cham)));
+                // TempDirectory path must end with a DirectorySeratorChar, otherwise it will throw in the Xml validation.
+                ss.Add(null, XmlReader.Create(new StringReader(main), null, tempDirectoryPath));
```

Comment:
But I'm not using XmlReaderSetting at all... Do I need to? And why only on Unix, OSX?